src/sys/bout_types.cxx

```diff
@@ -73,3 +81,88 @@ const std::string& DERIV_STRING(DERIV deriv) {
 
   return safeAt(DERIVtoString, deriv);
 }
+
+bool isXDirectionType(DIRECTION x) {
+  switch (x) {
+  case (DIRECTION::X):
+    return true;
+  default:
+    return false;
+  }
+}
+
+bool isYDirectionType(DIRECTION y) {
+  switch (y) {
+  case (DIRECTION::Y):
+  case (DIRECTION::YAligned):
+  case (DIRECTION::YOrthogonal):
+    return true;
+  default:
+    return false;
+  }
+}
+
+bool isZDirectionType(DIRECTION z) {
+  switch (z) {
+  case (DIRECTION::Z):
+    return true;
+  default:
+    return false;
+  }
+}
+
+bool compatibleDirections(DIRECTION d1, DIRECTION d2) {
```

Comment:
How about the following?

```cpp
return (isXDirectionType(d1) && isXDirectionType(d2))
    || (isYDirectionType(d1) && isYDirectionType(d2))
    || (isZDirectionType(d1) && isZDirectionType(d2))
    || (isSpecialDirectionType(d1) || isSpecialDirectionType(d1));
```

(needs `isSpecialDirectionType` to be implemented)

Reply:
Almost, need to add another condition so that we return `false` if `d1==YAligned` and `d2==YOrthogonal` or vice versa. Still think this looks nicer than the way I have it at the moment.

Reply:
actually, I'm not so sure about that. I want `YOrthogonal` to be compatible with `YOrthogonal` or `Y` but not `YAligned`, and `YAligned` to be compatible with `YAligned` or `Y` but not `YOrthogonal`. The single conditional gets a bit convoluted. Do you think it's still nicer than the nested switch-case statements?